Start announcement validation

avdata99 opened the pull request:
![image](https://user-images.githubusercontent.com/3237309/181074520-db5cf33f-93e5-4ec4-9c65-8c457f725daa.png)

![image](https://user-images.githubusercontent.com/3237309/181074615-3787a65b-b47b-4cb7-933a-57161400d4bc.png)

Other styles are already defined in #3
Backend also validate an raise error for incomplete data

amercader commented:
In this case I'd also try to reuse the standard form error handling in place in other CKAN form, so you don't need to worry about styling etc. Basically:

* You catch any `ValidationErrors` in the blueprint and create an `error_summary` dict with the form ` {"field_name": ["error1", "error2"]}` and pass this to the template when rendering (instead of redirecting)
* Add `{{ form.errors(error_summary) }}` to the top of the form

avdata99 commented:
Considering we down have a form pages for announcements (we have several modal forms), in the first step of the current version I added a flash message to show after the redirect:

![image](https://user-images.githubusercontent.com/3237309/181364517-c1607ad1-a842-44b3-8a7c-38631a846012.png)

As a second step, I added JS validation because we don't want to allow POSTing if some fields are empty.

Our [current validation ](https://github.com/okfn/ckanext-announcements/blob/validate_dates/ckanext/announcements/validators.py#L13) creates a ValidationError object which will generate the `error_sumary` automatically.

Should we stop using this modal forms and move to a _new_ and _edit_ form pages?
In the case we preserve this forms, is the current approach correct?

amercader commented:
This sounds like a good approach for now @avdata99
